lib/mini_profiler.rb

```diff
@@ -279,46 +220,23 @@ def call(env)
         status, headers, body = @app.call(env)
         client_settings.disable_profiling = true
         return client_settings.handle_cookie([status, headers, body])
-      else
-        client_settings.disable_profiling = false
       end
 
+      # remember that profiling is not disabled (ie enabled)
+      client_settings.disable_profiling = false
+
       # profile gc
       if query_string =~ /#{@config.profile_parameter}=profile-gc/
-        return tool_disabled_message(client_settings) if !advanced_debugging_enabled?
-        current.measure = false if current
-        return client_settings.handle_cookie(Rack::MiniProfiler::GCProfiler.new.profile_gc(@app, env))
+        return serve_profile_gc(env, client_settings)
```

Comment:
I think we should keep `current.measure = false` up at this level, it's not a view level concern.

Reply:
That makes sense, but it is slightly different logic. It's turning off measure _after_ serving the `tool_disabled_message`. It might be fine though, can see how tests like it.